src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -323,6 +314,17 @@ else if (payload != null)
             }
```

Comment:
This is where the `decode` loop should go.

In fact, given that all `MQTT` control packets have the same general layout, it would make sense to add the following structure to `protocol.idl`.
```
struct MqttPacketFixedHeader
{
    uint8 typeAndFlags;
    uint8 remainingLength;
}
```
and use it here to detect when there are sufficient bytes to decode a full `MqttPacket` including `remainingLength`.

Note: all `Mqtt*` packet structures in `idl` can `extend` `MqttPacketFixedHeader` to save repeating these common fields in each case.

If there are not sufficient bytes delivered to complete a successful `tryWrap` of `MqttPacketFixedHeader` or if there are not sufficient `remainingBytes`, then the partial bytes received so far should be copied to a buffer slot so they can be retained until the next `DATA` frame arrives.

When the next `DATA` frame arrives, if there is a `bufferSlot` in use, then the new `payload` bytes should be copied after the existing bytes in the `bufferSlot`, so you'll also need to track `slotLimit` and update it each time you copy bytes into the `bufferSlot` to know how much is in use.

Then, the `decodeState.decode(buffer, offset, length)` call will use the `bufferSlot` buffer instead of the `DataFW` `payload` buffer.

Note: there may be more than one packet to decode so you need to advance through the buffer until the `decoderState` is no longer making progress.

When the buffer has been decoded as far as possible, then the remaining bytes need to be copied to the beginning of the buffer slot.

If all the bytes have been consumed, then the `bufferSlot` (if any) should be returned to the `bufferPool`.